components/abstractions/src/main/java/com/microsoft/kiota/serialization/ParseNode.java

```diff
@@ -110,18 +111,18 @@ public interface ParseNode {
     /**
      * Gets the Enum value of the node.
      * @return the Enum value of the node.
-     * @param targetEnum the class of the enum.
+     * @param forValue a function that given a string returns the corresponding enum
      * @param <T> the type of the enum.
      */
-    @Nullable <T extends Enum<T>> T getEnumValue(@Nonnull final Class<T> targetEnum);
+    @Nullable <T extends Enum<T>> T getEnumValue(@Nonnull final Function<String, T> forValue);
 
     /**
      * Gets the EnumSet value of the node.
      * @return the EnumSet value of the node.
-     * @param targetEnum the class of the enum.
+     * @param forValue a function that given a string returns the corresponding enum
      * @param <T> the type of the enum.
      */
-    @Nullable <T extends Enum<T>> EnumSet<T> getEnumSetValue(@Nonnull final Class<T> targetEnum);
+    @Nullable <T extends Enum<T>> EnumSet<T> getEnumSetValue(@Nonnull final Function<String, T> forValue);
```

Comment:
I haven't found any usage of this method anywhere in the codebase nor in the generator repository.
`EnumSet` does have a very idiosyncratic API and the semantics of the method looks similar to `getCollectionOfEnumValues` AFAIU.
Given the [Occam's Razor](https://en.wikipedia.org/wiki/Occam%27s_razor) principle, should we consider removing it at all?

Reply:
The reason why enum sets are in the code base is to support "flaggable" enums. But there are limitations currently in the OAS spec and in the generation in tother languages, so we tabled it for a later major version.
see https://github.com/microsoft/kiota/issues/3238 and linked issue for the details.
We could leave it here for now, I don't think it hurts anyone even though it's not used today. If we add it back after GA, it'll be a breaking change with all the ceremony that's required, and our plans are not firm yet.

Reply:
Thanks for the explanation, I'm ok leaving it here, I doubt that `EnumSet` is going to be the final API, but we will see.